components/director/internal/domain/label/repository.go

```diff
@@ -0,0 +1,149 @@
+package label
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+)
+
+const tableName string = `"public"."labels"`
+const fields string = `"id", "tenant_id", "key", "value", "app_id", "runtime_id"`
+
+type dbRepository struct {
+}
+
+func NewRepository() *dbRepository {
+	return &dbRepository{}
+}
+
+func (r *dbRepository) Upsert(ctx context.Context, label *model.Label) error {
+	if label == nil {
+		return errors.New("Item cannot be empty")
+	}
+
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return errors.Wrap(err, "while fetching persistence from context")
+	}
+
+	entity, err := EntityFromModel(label)
+	if err != nil {
+		return errors.Wrap(err, "while creating Label entity from model")
+	}
+
+	stmt := fmt.Sprintf(`INSERT INTO %s (id, tenant_id, key, value, app_id, runtime_id) VALUES (:id, :tenant_id, :key, :value, :app_id, :runtime_id)
+		ON CONFLICT (id) DO UPDATE SET
+    		key = EXCLUDED.key,
+    		value = EXCLUDED.value,
+    		app_id = EXCLUDED.app_id,
```

Comment:
minor/theoretical: is it possible to upsert label and change app_id or runtime_id?

Reply:
Right, we shouldn't allow this kind of modifications.